app/routes/__init__.py

```diff
@@ -11,6 +11,7 @@
 
 DATASET_REGEX = r"^[a-z][a-z0-9_-]{2,}$"
 VERSION_REGEX = r"^v\d{1,8}(\.\d{1,3}){0,2}?$|^latest$"
+DATE_REGEX = r"^\d{4}(\-(0?[1-9]|1[012])\-(0?[1-9]|[12][0-9]|3[01]))?$"
```

Comment:
This is a broader version of the date regex which accepts (`yyyy`) year and full dates (`yyyy-mm-dd`). In most places only the full dates version was used. 
Let me think, if you see this as an issue, otherwise we might want to create entries here, once strict one and the one above accepting multiple inputs

Reply:
Doesn't strike me as an issue. I think this more flexible one is fine.